int32[] time
bool[] direction